app/controllers/coaches_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CoachesController < ApplicationController
+  def index
+  end
+
+  def show
+  end
+
+  private
+  helper_method :chapters
+  def chapters
+    @chapters ||= Chapter.by_name
+  end
+
+  helper_method :chapter
+  def chapter
+    @chapter ||= chapters.find_by!(slug: params[:chapter_slug])
+  end
+
+  helper_method :coaches
+  def coaches
+    @coaches ||= chapter.coaches.sort_by do |coach|
+      coach.attended_sessions.count
+    end.reverse
```

Comment:
I've moved this logic into the chapter model (I think it belongs there).

https://github.com/MikeRogers0/planner/blob/9aed586088fb5fd19943b828ee35ff4a772f1df9/app/controllers/dashboard_controller.rb#L30 - I started from here & found the DB to be pretty confusing to work with. 

I've not tried it with production levels of number, how many of members are there normally in a chapter?

> spending a load of time restructuring the data in the app (which definitely needs to be done at some point)

I was thinking adding the following fields:
`Member.coach` - boolean, updated when they attend a session as a coach.
`Member.sessions_attended_count` - A cache counter updated when sessions are attended.